src/runtime/contrib/dnnl/dnnl_json_runtime.cc

```diff
@@ -389,13 +463,238 @@ class DNNLJSONRuntime : public JSONRuntimeBase {
                          {DNNL_ARG_DST, out_memory}});
   }
 
+  void Concat(const size_t& nid) {
+    auto node = nodes_[nid];
+
+    std::vector<dnnl::memory::desc> data_mds;
+    std::vector<dnnl::memory> data_memories;
+    
+    auto data_entry = node.GetInputs()[0];
+    dnnl::memory::dims shape = nodes_[data_entry.id_].GetOpShape()[data_entry.index_];
+    int axis = std::stoi(node.GetAttr<std::vector<std::string>>("axis")[0]); // axis
+    
+    if (axis < 0) {
+        ICHECK_LE(0 - axis, shape.size());
+        axis = shape.size() + axis;
+    }
+    else ICHECK_LT(axis, shape.size());
+    
+    for (auto entry : node.GetInputs()) {
+      auto data_shape = nodes_[entry.id_].GetOpShape()[entry.index_];
+      dnnl::memory::desc data_md = GenDNNLMemDescByShape(data_shape, dt::f32);
+        data_mds.push_back(data_md);
+        data_memories.push_back(BindDNNLMemory(entry, data_md));
+      }
+    
+    auto concat_prim_desc = dnnl::concat::primitive_desc(axis, data_mds, engine_);
+    auto concat = dnnl::concat(concat_prim_desc);
+    net_.push_back(concat);
+    
+    auto out_md = concat_prim_desc.dst_desc();
+    JSONGraphNodeEntry out_entry(nid, 0);
+    auto out_memory = BindDNNLMemory(out_entry, out_md);
+
+    std::unordered_map<int, dnnl::memory> args;
+    for (int i = 0; i < (int)data_memories.size(); i++) {
+        args.insert({DNNL_ARG_MULTIPLE_SRC + i, data_memories[i]});
+    }
+    args.insert({DNNL_ARG_DST, out_memory});
+    net_args_.push_back(args);
+  }
+  
+  void MaxPool2d(const size_t& nid) {
+    auto node = nodes_[nid];
+
+    // Setup attributes.
+    auto data_entry = node.GetInputs()[0];
+    dnnl::memory::dims input_shape = nodes_[data_entry.id_].GetOpShape()[data_entry.index_];
+    dnnl::memory::dim pool_size0 = std::stoi(node.GetAttr<std::vector<std::string>>("pool_size")[0]); //notice
+    dnnl::memory::dim pool_size1 = std::stoi(node.GetAttr<std::vector<std::string>>("pool_size")[1]);
+    std::vector<std::string> str_strides = node.GetAttr<std::vector<std::string>>("strides");
+    std::vector<std::string> str_padding = node.GetAttr<std::vector<std::string>>("padding");
+    std::vector<std::string> ceil_mode = node.GetAttr<std::vector<std::string>>("ceil_mode");
+    dnnl::memory::dim dilation0 = std::stoi(node.GetAttr<std::vector<std::string>>("dilation")[0]);
```

Comment:
Does pooling have `dilation` option?

Reply:
Pooling does have dilation option.

Reply:
Does oneDNN also support this feature?

Reply:
OneDNN support dilation, and it can be found in onednn/examples/promitives/pooling.cpp.